play-services-core/src/main/kotlin/org/microg/gms/ui/SettingsFragment.kt

```diff
@@ -56,24 +60,30 @@ class SettingsFragment : ResourceSettingsFragment() {
     }
 
     private suspend fun updateDetails() {
-        if (getGcmServiceInfo(requireContext()).configuration.enabled) {
+        val context = requireContext()
+        val gcmServiceInfo = getGcmServiceInfo(context)
+        if (gcmServiceInfo.configuration.enabled) {
             val database = GcmDatabase(context)
             val regCount = database.registrationList.size
+            // check if we are connected as we should be and re-connect if not
+            if (!McsService.isConnected(context)) {
```

Comment:
I think `McsService` is in the persistent process, so we can't/shouldn't do this check here like this, right?

Reply:
I changed this to use the `gcmServiceInfo` directly.

Reply:
Thanks